map.js

```diff
@@ -46,51 +46,90 @@ legend.onAdd = function() {
 };
 legend.addTo(map);
 
+/*
+fetch (states10mSource)
+  .then(response => response.json())
+  .then(function(json) {
+      L.vectorGrid.slicer(json, {
+        rendererFactory: L.canvas.tile,
+        vectorTileLayerStyles: {
+          sliced: function (properties, zoom) {
+            return stateStyle(properties);
+          }
+        }
+      }).addTo(map);
+  });
+*/
 const statesLayer = new ZoomShowHide().addTo(map);
 
 const ZOOM_THRESHOLD = 5;
 
-const states50m = L.geoJSON(null, {
-  style: stateStyle
-});
-statesLayer.addLayer(states50m);
+let states50m = '';
+
+//  statesLayer.addLayer(states50m);
+
+let states10m = '';
 
-const states10m = L.geoJSON(null, {
-  style: stateStyle
-});
-states10m.min_zoom = ZOOM_THRESHOLD + 1;
-statesLayer.addLayer(states10m);
+//  states10m.min_zoom = ZOOM_THRESHOLD + 1;
+//  tatesLayer.addLayer(states10m);
 
+/*
 function loadTopoToLayer(topology, layer) {
   const geojson = topojson.feature(topology, topology.objects.countries);
-  layer.addData(geojson);
+  layer.options.dataSource = geojson;
   addLayerInteraction(layer);
 }
+*/
 
 fetch(states50mSource)
   .then(response => response.json())
-  .then(topology => loadTopoToLayer(topology, states50m));
+  .then(function(json) {
+      states50m = L.vectorGrid.slicer(json, {
+        rendererFactory: L.canvas.tile,
+        vectorTileLayerStyles: {
+          countries: function(properties) {
+            return stateStyle(properties);
+          }
+        }
+      }).addTo(map);
+  });
 
 function zoomHandler() {
   const currentZoom = map.getZoom();
   if (currentZoom < ZOOM_THRESHOLD) return;
   map.off('zoomend', zoomHandler);
   fetch(states10mSource)
     .then(response => response.json())
-    .then(topology => loadTopoToLayer(topology, states10m))
-    .then(() => {
-      states50m.max_zoom = ZOOM_THRESHOLD;
-      statesLayer.filter();
-    });
+    .then(function(json) {
+        states10m = L.vectorGrid.slicer(json, {
+          rendererFactory: L.canvas.tile,
+          vectorTileLayerStyles: {
+            countries: function(properties) {
```

Comment:
You don't need to set the `countries` property to an anonymous function, you can set it to the `stateStyle` function directly:
```js
vectorTileLayerStyles: {
  countries: stateStyle
}
```

Reply:
I need to work with "properties" of the layer, so I don't think this notation would work... I proposed something different, but the Styling issue is still not resolved.

Reply:
set to `countries: stateStyle` as proposed